applications/nrf_desktop/src/modules/fast_pair.c

```diff
@@ -48,18 +50,33 @@ static bool in_pairing_mode(uint8_t bt_local_id)
 
 static void update_fast_pair_ui_indication(void)
 {
-	bool show_ui_pairing = can_pair(local_id);
+	bool show_ui_pairing = can_pair(selected_stack_id);
 
 	bt_le_adv_prov_fast_pair_show_ui_pairing(show_ui_pairing);
 
-	if (in_pairing_mode(local_id)) {
+	if (in_pairing_mode(selected_stack_id)) {
 		LOG_INF("Fast Pair discoverable advertising");
 	} else {
 		LOG_INF("%s Fast Pair not discoverable advertising UI indication",
 			(show_ui_pairing) ? ("Show") : ("Hide"));
 	}
 }
 
+static void control_fast_pair_provider(uint8_t app_id, uint8_t stack_id)
+{
+	bool fp_provider_enabled = true;
+
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_DONGLE_PEER_ID_INFO)) {
+		fp_provider_enabled = (app_id != dongle_app_id);
+		bt_le_adv_prov_fast_pair_enable(fp_provider_enabled);
+	}
+
+	if (fp_provider_enabled) {
+		selected_stack_id = stack_id;
```

Comment:
Consider updating this field always (to keep the information updated in module).

Reply:
You can even call `update_fast_pair_ui_indication` no matter if provider is enabled (actually you do it when handling `ble_peer_event`). If provider is disabled, it would not affect advertising payload anyway.